components/eam/src/chemistry/modal_aero/modal_aero_amicphys_diags.F90

```diff
@@ -0,0 +1,168 @@
+module modal_aero_amicphys_diags
+
+  implicit none
+
+  public
+
+contains
+
+!-----------------------------------------------------------------------------------
+! Purpose: set switches that turn on or off column-integrated tendency diagnostics.
+!-----------------------------------------------------------------------------------
+subroutine amicphys_diags_init( do_cond, do_rename, do_newnuc, do_coag )
+
+  use modal_aero_amicphys_control, only: do_q_coltendaa, do_qqcw_coltendaa, &
+                                         iqtend_cond, iqtend_rnam, iqtend_nnuc, iqtend_coag, &
+                                         iqqcwtend_rnam
+
+  logical,intent(in) :: do_cond, do_rename, do_newnuc, do_coag
+
+  if ( (.not. do_cond) .and. (.not. do_rename) ) then
+     do_q_coltendaa(:,iqtend_cond) = .false.
+     do_q_coltendaa(:,iqtend_rnam) = .false.
+     do_qqcw_coltendaa(:,iqqcwtend_rnam) = .false.
+  end if
+  if (.not.do_newnuc) do_q_coltendaa(:,iqtend_nnuc) = .false.
+  if (.not.do_coag)   do_q_coltendaa(:,iqtend_coag) = .false.
+
+end subroutine amicphys_diags_init
+
+!--------------------------------------------------------------------------------
+! Purpose: Get grid cell mean tendencies by calculating area-weighted averages
+!          of the values in different subareas.
+!--------------------------------------------------------------------------------
+subroutine get_gcm_tend_diags_from_subareas( nsubarea, ncldy_subarea, afracsub, &! in
+                                             qsub_tendaa, qqcwsub_tendaa,       &! in
+                                             qgcm_tendaa, qqcwgcm_tendaa        )! out
+
+  use modal_aero_amicphys_control, only: wp=>r8, ncnst=>gas_pcnst, maxsubarea, nqtendaa, nqqcwtendaa
```

Comment:
I don't know for sure. My impression is that `aa` is used for distinguishing those variables from another set of variables ending in `bb`. Those `bb` variables seems to be used only in the box model. Both the `aa` and `bb` variables are used for diagnostics only. 

I am guessing that, at some point, somebody will need to decide how to organize the diagnostics in MAMXX. At that point, it would be useful to organize those `aa` variables in to a data structure or replace `aa` with `_diag` or something similar.

A related thought: I am guessing the BFB testing mechanism we are using for the Fortran code refactoring does not cover the aerosol diagnostics (i.e., those calculations that are used only when `history_aerosol = .t.`, `history_verbose = .t.` etc. and do not affect the time integration). Is that correct? If so, we would want to keep in mind that at a later point, those diagnostics need to be revisited.